test/utils/integration/integration.go

```diff
@@ -873,6 +941,37 @@ func (env TestEnvironment) GetServiceFromK8s(name, namespace string) (*corev1.Se
 	return result, nil
 }
 
+func (env TestEnvironment) GetSecretFromK8s(name, namespace string) (*corev1.Secret, error) {
+	nn := types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}
+	result := &corev1.Secret{}
+	if err := env.k8sClient.Get(env.Context, nn, result); err != nil {
+		return nil, err
+	}
+	return result, nil
+}
```

Comment:
nitpick: maybe create a common `GetSecret` function in KubeClient, which can be used by `func (r *Reconciler) reconcileWebhooksWithCABundle(ctx context.Context) error` in weebhook.go

Reply:
updated it to use the common method from `KubeClient`.